src/pkg/oci/pull.go

```diff
@@ -136,9 +140,17 @@ func (o *OrasRemote) PullPackage(destinationDir string, concurrency int, layersT
 
 	estimatedBytes := int64(0)
 	if isPartialPull {
-		for _, path := range AlwaysPull {
-			desc := manifest.Locate(path)
-			layersToPull = append(layersToPull, desc)
+		for _, path := range PackageAlwaysPull {
+			exists := false
+			for _, layer := range layersToPull {
+				if layer.Annotations[ocispec.AnnotationTitle] == path {
+					exists = true
+				}
+			}
+			if !exists {
+				desc := manifest.Locate(path)
+				layersToPull = append(layersToPull, desc)
+			}
```

Comment:
this needs to use similar SHA logic as implemented later. As it stands you could have duplication of paths, and the wrong size is used, resulting in a wrong progress bar.